### PR TITLE
Improve docs for first time developer setup

### PR DESCRIPTION
**Pull request by Steve-Mcl**
Improve docs for first time developer setup...
* adds a TOC with links to sections for simple stepwise "how to"
* tested on windows by following the steps to the letter
* adds VSCode tips for step debugging
* adds link from front page README to the main contribute documentation

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -21,15 +21,17 @@ npm install
...
 npm start
 ```
 
-If you're actively developing, then use `npm run serve` to run the application
+If you're actively developing for the FlowForge core, then use `npm run serve` to run the application
 with automatic rebuilds of the front-end, and restarts of the application when
 it is modified.
 
+If you intend on developing for the additional repositories, please refer to [the docs](https://github.com/flowforge/flowforge/tree/dev-docs#readme) 
+for detailed developer setup instructions
```
Comment (ZJvandeWeg) [inline README.md:33]:
```suggestion
for detailed developer setup instructions.
```

File: docs/contribute/README.md [hunk 2]
```diff
@@ -41,7 +42,7 @@ want to begin.
 
 ### FlowForge Code Structure
 
-```
+```console
```
Comment (ZJvandeWeg) [inline docs/contribute/README.md:45]: Can't use console here. You break syntax highlighting. Please remove it

Comment (Steve-Mcl) [reply docs/contribute/README.md:45]: Odd, works for me! Will remove.

**Comment by hardillb** [comment]
@Steve-Mcl The list of supported languages for the rendered used for the website (not VSCode)

https://lucidar.me/en/web-dev/list-of-supported-languages-by-prism/

File: README.md [hunk 3]
```diff
@@ -21,15 +21,17 @@ npm install
...
 # Start the application
 npm start
 ```
 
-If you're actively developing, then use `npm run serve` to run the application
+If you're actively developing for the FlowForge core, then use `npm run serve` to run the application
 with automatic rebuilds of the front-end, and restarts of the application when
 it is modified.
 
+If you intend on developing for the additional repositories, please refer to [the docs](https://github.com/flowforge/flowforge/tree/dev-docs#readme)
```
Comment (knolleary) [inline README.md:32]:
```suggestion
If you intend on developing for the additional repositories, please refer to [the docs](https://github.com/flowforge/flowforge/tree/dev-docs#readme) 
```
Where should this be linked to? Currently it points at the same file but on your feature branch (that will get deleted once this PR is merged).

File: docs/contribute/README.md [hunk 4]
```diff
@@ -51,66 +52,85 @@ want to begin.
...
+1. [Mocking email](#mocking-email)
+1. [Testing](#testing)
+1. [VSCode Tips](#vscode-tips)
 
+### Clone the `flowforge/flowforge` repository
 
-After cloning the core repository, you will need to install the dependencies by running
-```
-npm install
+```console
```
Comment (ZJvandeWeg) [inline docs/contribute/README.md:67]: Please don't use console, our syntax highlighting breaks with this.

File: docs/contribute/README.md [hunk 5]
```diff
@@ -51,66 +52,85 @@ want to begin.
...
-```
+#### OPTION 1 NPM
 
-In the `flowforge-nr-audit-logger` directory run
-```
+
+After cloning the core repository, you will need to install 
+`flowforge/flowforge` dependancies
+
+```console
```
Comment (ZJvandeWeg) [inline docs/contribute/README.md:92]:
```suggestion
```
```

File: docs/contribute/README.md [hunk 6]
```diff
@@ -51,66 +52,85 @@ want to begin.
...
+
+#### OPTION 2 Source Code
+
+After cloning the core repository, you will need to install 
+`flowforge/flowforge` dependancies
+
+Instead of using NPM, you can instead run from the latest source code.
+You can check out all the required projects in the same directory
+
+```console
```
Comment (ZJvandeWeg) [inline docs/contribute/README.md:109]:
```suggestion
```
```

File: docs/contribute/README.md [hunk 7]
```diff
@@ -51,66 +52,85 @@ want to begin.
...
+`flowforge`, `flowforge-nr-launcher` and `flowforge-driver-localfs` projects. 
+> DO NOT check modifications into git.
 
 ### Create a Stack
 You will need to setup the version(s) of Node-RED you want to use in your stacks.
 
 From the flowforge directory run
 
-```
+```console
```
Comment (ZJvandeWeg) [inline docs/contribute/README.md:133]:
```suggestion
```
```

File: docs/contribute/README.md [hunk 8]
```diff
@@ -51,66 +52,85 @@ want to begin.
...
 
-**Note** If running on MacOS 12.3 or newer you may get an error around node-gyp being unable to build sqllite3.
-This is because MacOS no longer includes python2.7.
-The solution is to run the command `npm config set python python3` to alias to python3 and then run `npm install` again
+### Install `flowforge/flowforge` dependencies
 
-By default this will install the latest released versions of the FlowForge components. If
-you want to run from the latest source code then you can check out all the required 
-projects in the same directory
+Once the core project is cloned, you will need to install its depenedancies.
```
Comment (knolleary) [inline docs/contribute/README.md:73]:
```suggestion
Once the core project is cloned, you will need to install its dependencies.
```

**Review by knolleary — CHANGES_REQUESTED** [review]

**Comment by ZJvandeWeg** [comment]
@Steve-Mcl Other than the nit picks I found, feel free to move forward with these changes!

File: README.md (hunk 3 above)
Comment (Steve-Mcl) [reply README.md:32]:
```suggestion
If you intend on developing for the additional repositories, please refer to [the docs](https://flowforge.com/docs/contribute/) 
```

Well spotted. Should be `https://flowforge.com/docs/contribute/`

Comment (Steve-Mcl) [reply README.md:32]: Or should it be `https://github.com/flowforge/flowforge/tree/main/docs/contribute#readme`? 🤔

**Comment by Steve-Mcl** [comment]
@ZJvandeWeg @knolleary fixed up the niggles pointed out (thanks for the set of eyes)

While on spell checking, I found a few other typos & since being in the docs, I fixed these up too.

If you guys are happy, I'll merge.

**Comment by Steve-Mcl** [comment]
@ZJvandeWeg @knolleary 

I needed a fresh dev platform and found a small issue (now resolved in 5449f10)

I tested the instructions 3 times in a row on windows. Working every time.

I have also tested on a PI and WSL (ubuntu) - working also.

Please (re)review / merge when time permits.

**Review by knolleary — APPROVED** [review]

**Pull request merged**